src/mjolnir/pbfgraphparser.cc

```diff
@@ -106,10 +106,12 @@ struct graph_callback : public OSMPBF::Callback {
       return;
     }
 
-    // Get tags if not already available
-    results = results ? results : lua_.Transform(OSMType::kNode, tags);
-    if (results->size() == 0) {
-      return;
+    // Get tags if not already available.  Don't bother calling Lua if there
+    // are no OSM tags to process.
+    if (tags.size() > 0) {
+      results = results ? results : lua_.Transform(OSMType::kNode, tags);
+    } else {
+      results = results ? results : Tags();
```

Comment:
seems like from the line below we need to work on the defaults in the constructor for `Tags` and basically have it do whatever lua normal does when it sees no tags? actually you know what we can do? we can add to the constructor of the graphparser a quick call to `lua_.Transform()` for each type of thing (node way relation) with an empty set of tags and store the result to use here. That way we dont have to figur eout how to replicate the behavior inside of lua in c++ (although i reckon one day we should just completley dump lua, its just today is not the day).

Reply:
so concretely inside of the `graph_callback` constructor we can do:

```c++
empty_node_results_ = lua_.Transform(OSMType::kNode, {});
empty_way_results_ = lua_.Transform(OSMType::kWay, {});
empty_relation_results_ = lua_.Transform(OSMType::kRelation, {});
```

(probably we dont need way or relation because when they are untagged we ignore them completely. the last time i worked on this code was years ago and i didnt scroll down to find out :smile: )

Reply:
I did this - it works great.